src/app/server-cluster/ServerClusterInterfaceRegistry.cpp

```diff
@@ -0,0 +1,288 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/server-cluster/ServerClusterInterfaceRegistry.h>
+
+#include <app/ConcreteClusterPath.h>
+#include <app/server-cluster/ServerClusterInterface.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/CodeUtils.h>
+
+namespace chip {
+namespace app {
+
+namespace {
+
+/// Remove all elements of a linked list rom the linked list.
+///
+/// Marks every element in the given linked list as not being in a
+/// linked list anymore.
+void ClearSingleLinkedList(ServerClusterInterface * clusters)
+{
+    while (clusters != nullptr)
+    {
+        ServerClusterInterface * next = clusters->GetNextListItem();
+        clusters->SetNotInList();
+        clusters = next;
+    }
+}
+
+ServerClusterInterfaceRegistry sRegistry;
```

Comment:
This should not be a singleton. This should belong to the data model provider that wants to register.

Reply:
Moved it away from a singleton: plan to have the registry itself then be the registry.